src/Streamly/Data/Fold.hs

```diff
@@ -6,16 +6,15 @@
 -- Stability   : experimental
 -- Portability : GHC
 --
--- A 'Fold' is a sink or a consumer of a stream of values.  The 'Fold' type
--- consists of an accumulator and an effectful action that absorbs a value into
--- the accumulator.
+-- A 'Fold' is an abstraction that can act as a consumer for a stream of values.
+--
+-- A fold driver e.g. 'Streamly.Prelude.fold' pushes values from a stream to the
```

Comment:
`driver` is an important part of the design. We also have `fold_` now. We should describe this as soon as possible in the doc.